services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java

```diff
@@ -1180,7 +1180,6 @@ public static ClientSessionContext attachSession(AuthenticationSessionModel auth
         }
 
         event.user(userSession.getUser())
-                .detail(Details.USERNAME, username)
```

Comment:
This is an example where the username should be stored as entered by the user, as we don't want to change the logic of existing events. It might sometimes be important to know what the user actually entered in the form. 

So this line and similar ones should be restored.